models/__tests__/bonus_activation.test.ts

```diff
@@ -21,10 +22,10 @@ const aCollectionUri = DocumentDbUtils.getCollectionUri(
 );
 
 const aFiscalCode = "AAABBB80A01C123D" as FiscalCode;
-const aBonusId = "bonusid123" as NonEmptyString;
-const aBonusActivationId = `${aFiscalCode}-${aBonusId}` as NonEmptyString;
+const aBonusId = "AAAAAAAAAAAA" as NonEmptyString & BonusCode;
```

Comment:
@gunzip I had to cast this to an intersection between `NonEmptyString & BonusCode` or else the model would not accept this value for the `id` attribute as it's actually an intersection between `id: BonusCode` and `id: NonEmptyString` from [here](https://github.com/pagopa/io-functions-bonus/blob/master/models/bonus_activation.ts#L28)

Reply:
actually is the same for `FiscalCode`. Should we assume pattern strings are non-empty?

Reply:
@balanza  I was just thinking about that, we can't assume that, since `^$` is a pattern for empty strings